Makefile

```diff
@@ -9,18 +9,32 @@ IMAGE_NAME := node-feature-discovery
 IMAGE_TAG_NAME := $(VERSION)
 IMAGE_REPO := $(IMAGE_REGISTRY)/$(IMAGE_NAME)
 IMAGE_TAG := $(IMAGE_REPO):$(IMAGE_TAG_NAME)
-
+K8S_NAMESPACE := kube-system
 
 all: image
 
-image:
+image: nfd-master.yaml nfd-worker-daemonset.yaml
 	$(IMAGE_BUILD_CMD) --build-arg NFD_VERSION=$(VERSION) \
 		-t $(IMAGE_TAG) ./
 
+%.yaml: %.yaml.template FORCE
+	@echo "$@: namespace: ${K8S_NAMESPACE}"
+	@echo "$@: image: ${IMAGE_TAG}"
+	@sed -E \
+	     -e s',^(\s*)name: node-feature-discovery,\1name: ${K8S_NAMESPACE},' \
+	     -e s',^(\s*)image:.+$$,\1image: ${IMAGE_TAG},' \
+	     -e s',^(\s*)namespace:.+$$,\1namespace: ${K8S_NAMESPACE},' \
+	     $< > $@
+
+docker-push: image
+	docker push "${IMAGE_TAG}"
```

Comment:
I have a corresponding addition in #160. We support other image builders than docker, too, so I think the make target should be `image-push`. Also, the push command should be configurable, e.g. `$(IMAGE_PUSH_CMD)`

Reply:
As #160 seems to be going in, I dropped `docker-push` completely from this patch.
(I copied `docker-push:` practice from projects created with `kubebuilder`.)
Changed `README.md` examples to use `make push`, that is implemented in #160.